src/Module.ts

```diff
@@ -439,9 +466,24 @@ export default class Module {
 
 		// we don't want to create shims when we are just
 		// probing export * modules for exports
-		if (this.graph.shimMissingExports && !isExportAllSearch) {
-			this.shimMissingExport(name);
-			return this.exportShimVariable;
+		if (!isExportAllSearch) {
+			if (this.syntheticNamedExports) {
+				const syntheticExport = this.syntheticExports.get(name);
+				if (syntheticExport) {
+					return syntheticExport;
+				}
+				if (!this.exports[name]) {
```

Comment:
Still wondering about this check. Shouldn’t L450-L456 make sure that this condition is always fulfilled because otherwise we would have done an early return there?

Reply:
ups! done